connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkResult.scala

```diff
@@ -80,7 +80,10 @@ private[sql] class SparkResult[T](
           }
           while (reader.loadNextBatch()) {
             val rowCount = root.getRowCount
-            assert(root.getRowCount == response.getArrowBatch.getRowCount) // HUH!
+            assert(
+              response.getIsLocalBuilt ||
+                root.getRowCount == response.getArrowBatch.getRowCount
```

Comment:
I'd just make the row count optional. It is kinda weird it is in the protocol anyway.

Reply:
You means change the protocol that make the row count optional ?

Reply:
```
Error: Field "1" on message "ArrowBatch" moved from outside to inside a oneof.
[8](https://github.com/beliefer/spark/actions/runs/4761585011/jobs/8462987474#step:5:9)
Error: buf found 1 breaking changes.
```

It seems 1 breaking change.

Reply:
I guess adding the row count as optional to `LocalRelation` message is more useful.
The server can check if the expected number of rows are provided, and we can reuse it here to set the row count.

Reply:
@hvanhovell

Reply:
@hvanhovell What do you think about my idea if making the row count optional causes the breaking change?

Reply:
@hvanhovell Could you take a close look ?